### PR TITLE
Feature/add identity server

### LINKED ISSUE #3: User registration endpoint
To do:

- [ ] Registration endpoint
- [ ] Send verification email
- [ ] Validate email endpoint

### PR DESCRIPTION
**Pull request by idormenco**
Configured identity server:
* added initial migrations for the identity database
* only the swagger UI client is configured for now (added the `Authorize` attribute to the `backpack` controller, to test)
* configured authentication schemes in the Api project

Tested locally (swagger ui>authorize>call backpack GET endpoint>i get the user claims in the `User` object)

Closes #3

### CODE REVIEW

File: Src/DeUrgenta.IdentityServer/appsettings.json [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            "Microsoft.Hosting.Lifetime": "Information"
+        }
+    },
+    "AllowedHosts": "*",
+    "ConnectionStrings": {
+        "ApplicationDbContextConnection": "Server=localhost;Port=5432;Database=de-urgenta-db;User Id=docker;Password=docker;"
+    },
+    "ApiConfiguration": [
+        {
+            "Name": "answersApi",
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/appsettings.json]: cleanup

File: Src/DeUrgenta.IdentityServer/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            "Secret": "svpqYnJSR8xzn8Rl"
+        },
+        {
+            "Name": "usersApi",
+            "Secret": "st4k!b7s$af201cv",
+            "ClaimList": [ "openid", "email" ]
+        }
+    ],
+    "ClientApplications": [
+        {
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/appsettings.json]: cleanup this list: maybe just keep the swagger client for now?

File: Src/DeUrgenta.IdentityServer/appsettings.json [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                "email",
+                "answersApi",
+                "usersApi"
+            ],
+            "AllowAccessTokensViaBrowser": true,
+            "AccessTokenType": 1
+        }
+    ],
+    "EMailingSystem": "SendGrid",
+    "AdminFromName": "Echipa Jurnal Medical",
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/appsettings.json]: rename

File: Src/DeUrgenta.IdentityServer/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        "Password": "1234"
+    },
+    "AllowedRedirectUrls": [
+        "https://prod.stamacasa.ro/register-complete"
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/appsettings.json]: cleanup

File: Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/stateEntityTemplate.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
+﻿<!doctype html>
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/stateEntityTemplate.html]: remove this file

File: Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/resetPasswordTemplate.html [hunk 6]
```diff
@@ -0,0 +1,32 @@
+﻿<!doctype html>
+<html>
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/resetPasswordTemplate.html:2]: We should review this template.

File: Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/dailyAssessmentTemplate.html [hunk 7]
```diff
@@ -0,0 +1,34 @@
+﻿<!doctype html>
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/dailyAssessmentTemplate.html]: remove file

File: Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/dailyAssessmentTemplate.html [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+<html>
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/dailyAssessmentTemplate.html]: remove file

File: Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/accountConfirmationTemplate.html [hunk 9]
```diff
@@ -0,0 +1,40 @@
+﻿<!doctype html>
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/Resources/accountConfirmationTemplate.html:1]: review content for this template

File: Src/DeUrgenta.IdentityServer/Services/Emailing/EmailTemplate.cs [hunk 10]
```diff
@@ -0,0 +1,10 @@
+﻿namespace DeUrgenta.IdentityServer.Services.Emailing
+{
+    public enum EmailTemplate
+    {
+        AccountConfirmation,  
+        DailyAssessment,
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/EmailTemplate.cs:6]:
```suggestion
```

File: Src/DeUrgenta.IdentityServer/Services/Emailing/EmailTemplate.cs [hunk 11]
```diff
@@ -0,0 +1,10 @@
+﻿namespace DeUrgenta.IdentityServer.Services.Emailing
+{
+    public enum EmailTemplate
+    {
+        AccountConfirmation,  
+        DailyAssessment, 
+        StateEntity,
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/EmailTemplate.cs:7]:
```suggestion
```

File: Src/DeUrgenta.IdentityServer/Services/Emailing/EmailConstants.cs [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+        public static string GetTemplatePath(EmailTemplate template)
+        {
+            return templatePaths[template];
+        }
+
+        private static Dictionary<EmailTemplate, string> subjects = new Dictionary<EmailTemplate, string>()
+        {
+            { EmailTemplate.AccountConfirmation, "Bine ai venit!"},
+            { EmailTemplate.DailyAssessment, "Un minut pe zi pentru o viață liniștită. Chestionar zilnic de evaluare"},
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/EmailConstants.cs:20]:
```suggestion
```

File: Src/DeUrgenta.IdentityServer/Services/Emailing/EmailConstants.cs [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+            { EmailTemplate.StateEntity, "Rezultate evaluare persoane autoizolare"},
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/EmailConstants.cs:21]:
```suggestion
```

File: Src/DeUrgenta.IdentityServer/Services/Emailing/EmailConstants.cs [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+            { EmailTemplate.ResetPassword, "Resetare parolă"}
+        };
+
+        private static Dictionary<EmailTemplate, string> templatePaths = new Dictionary<EmailTemplate, string>()
+        {
+            { EmailTemplate.AccountConfirmation, "accountConfirmationTemplate.html"},
+            { EmailTemplate.DailyAssessment, "dailyAssessmentTemplate.html"},
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/EmailConstants.cs:28]:
```suggestion
```

File: Src/DeUrgenta.IdentityServer/Services/Emailing/EmailConstants.cs [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+            { EmailTemplate.StateEntity, "stateEntityTemplate.html"},
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Services/Emailing/EmailConstants.cs:29]:
```suggestion
```

File: Src/DeUrgenta.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+                    </div>
+                    <p>Parola trebuie să includă cel puţin o literă mică, o literă mare, o cifră şi un simbol.</p>
+
+                    <br />
+                    <div class="field">
+                        <div class="container __login-form-options">
+                            <label class="__checkbox-label" asp-for="Input.AcceptTermsAndConditions">
+                                <input type="checkbox" asp-for="Input.AcceptTermsAndConditions">
+                                <span class="__checkbox-custom"></span>
+                                Declar că am citit și sunt de acord cu <a href="https://jurnalmedical.ro/termeni-si-conditii">Termenii și Condițiile</a> aplicației Jurnal Medical.
```
Comment (trupci) [inline Src/DeUrgenta.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml]: review link

**Review by trupci — CHANGES_REQUESTED** [review]

File: Src/DeUrgenta.IdentityServer/appsettings.json (hunk 1 above)
Comment (trupci) [reply Src/DeUrgenta.IdentityServer/appsettings.json]: and configure scopes for the new apis

**Review by Puiu — CHANGES_REQUESTED** [review]
what @trupci said :)

**Review by catileptic — APPROVED** [review]

**Pull request merged**